### PR TITLE
Fixed syntax.

### PR DESCRIPTION
**Pull request by click0**

### CODE REVIEW

**Comment by MarvAmBass** [comment]
Hi there, thanks for the request - looks good - the only thing I don't understand is the forking systemd option. 

Why did you set it here?

Thanks 

Marvin

File: templates/socat.service.j2 [hunk 1]
```diff
@@ -1,8 +1,10 @@
 [Unit]
 Description=socat tunnel service {{ socat_service_name }}
-After=network.target
+After=network-online.target
+Wants=network-online.target
 
 [Service]
+Type=forking
```
Comment (click0) [inline templates/socat.service.j2:7]: I refer to `socat` as a traditional UNIX service and still hope to be able to add the PID file at startup.
https://www.freedesktop.org/software/systemd/man/systemd.service.html#Options

File: templates/socat.service.j2 [hunk 2]
```diff
@@ -1,8 +1,10 @@
 [Unit]
 Description=socat tunnel service {{ socat_service_name }}
-After=network.target
```
Comment (click0) [inline templates/socat.service.j2:3]: I tested the role on the new Debian 11.2
There is `network-online.target`, not `network.target`.
You may have to do `get_facts` and insert the necessary units depending on the OS ...

**Comment by click0** [comment]
> Hi there, thanks for the request - looks good - the only thing I don't understand is the forking systemd option.
> 
> Why did you set it here?
> 
Hi.
I've added some clarification to review, please answer there.

File: templates/socat.service.j2 (hunk 1 above)
Comment (MarvAmBass) [reply templates/socat.service.j2:7]: yeah but if you read about the `forking`-type - it states it's for services which will fork into background - as far as I know socat usually stays in foreground and can therefore be controlled by systemd directly.

I'm not sure if there are socat options to force it into background - there might be -> maybe we should only add this if the configuration matches a certain (background forking) pattern

part from the forking section:
```
The parent process is expected to exit when start-up is complete and all communication channels are set up.
```

File: templates/socat.service.j2 (hunk 2 above)
Comment (MarvAmBass) [reply templates/socat.service.j2:3]: so debian prior 11 didn't support `network-online.target` and only had `network.target`? that's good to know! thx!

**Comment by MarvAmBass** [comment]
@click0 thanks for helping me - and I'm sorry that I ask so many questions, but I want to be a reliable source for my users - so I need to double check every merge and clarify that there are no risk for side-effects, errors etc.

**Comment by click0** [comment]
> @click0 thanks for helping me - and I'm sorry that I ask so many questions, but I want to be a reliable source for my users - so I need to double check every merge and clarify that there are no risk for side-effects, errors etc.

Your role is not present in ansible-galaxy, and it's too early to tell if anyone is using the role at all. 
You should always check on your test environment.
The role is not ideal, the system can have all sorts of services and tunnels that need socat. More likely, additional firewall rules (iptables) will be needed, but for now I use them in the context of the playbook.
You can always rename the role, for example `ansible-role-socat@stable` and use the stable branch, and test innovations in the dev branch :) 

Also, in the future I will have to add FreeBSD support for the role or make a separate role for it ....

File: templates/socat.service.j2 (hunk 1 above)
Comment (MarvAmBass) [reply templates/socat.service.j2:7]: I'll merge and switch this back to simple

File: templates/socat.service.j2 (hunk 2 above)
Comment (MarvAmBass) [reply templates/socat.service.j2:3]: just read into that and learned something new ;)

**Comment by click0** [comment]
In general, you are not obligated to merge a pull request.
You don't have to close it, and my further edits will be added to the pull-request themselves.

**Pull request merged**